blog/02-01-2023-flashloan-aave-celo/flashloan aave celo.md

```diff
@@ -0,0 +1,254 @@
+
+We will limit access to it to the owner (i.e. you), this will help guard against griefing attacks. You will request a Flash Loan of 1 USDC (1* 10^6 as USDC has 6 decimal places). We don't need data for the flashloan, so we are passing an empty string. 
+
+This function then calls the flashloan() function on the lending pool contract for you to receive the tokens.
+```
+    function myFlashLoanCall(address _tokenAddress) public onlyOwner {
+        address receiverAddress = address(this);
+
+        address[] memory assets = new address[](1);
+        assets[0] = address(_tokenAddress); //goerli USDC 0x9FD21bE27A2B059a288229361E2fA632D8D2d074
+
+        uint256[] memory amounts = new uint256[](1);
+        amounts[0] = 1*10**6;
+
+        // 0 = no debt, 1 = stable, 2 = variable
+        uint256[] memory modes = new uint256[](1);
+        modes[0] = 0;
+
+        address onBehalfOf = address(this);
+        bytes memory params = "";
+        uint16 referralCode = 0;
+
+        LENDING_POOL.flashLoan(
+            receiverAddress,
+            assets,
+            amounts,
+            modes,
+            onBehalfOf,
+            params,
+            referralCode
+        );
+    }
+}
+```
+
+## Step 3: Executing your Flash Loan operation
+If we have requested a valid amount from a valid reserve, then the LendingPool contract will call the executeOperation() function in our contract. This is where we utilize the funds received from the lending pool. It checks if we received the appropriate loan amount or will throw an error message.
+As shown below, we can include our executeOperation() function:
+```
+    function executeOperation(
+        address[] calldata assets,
+        uint256[] calldata amounts,
+        uint256[] calldata premiums,
+        address initiator,
+        bytes calldata params
+    )
+        external
+        override
+        returns (bool)
+    {
+
+        //
+        // This contract now has the funds requested.
+        // Your logic goes here.
+        //
+
+        // At the end of your logic above, this contract owes
+        // the flashloaned amounts + premiums.
+        // Therefore ensure your contract has enough to repay
+        // these amounts.
+
+        // Approve the LendingPool contract allowance to *pull* the owed amount
+        for (uint i = 0; i < assets.length; i++) {
+            uint amountOwing = amounts[i].add(premiums[i]);
+            IERC20(assets[i]).approve(address(LENDING_POOL), amountOwing);
+        }
+
+        return true;
+    }
+
+```
+Before the end of the execution, you must approve the Lending pool to pull (payback) the borrowed funds. Failure to do so will cause the transaction to revert.
+
+Kindly note that the executeOperation() function signature must match exactly. When the LendingPool contract calls your executeOperation() function, it will pass in the assets, amounts, premiums, initiator and params for you to use in your operation.
+```
```

Comment:
add a space before the code snippet and add the intended code language like this 
" ```solidity
// code here
``` "